bitcoin/src/sighash.rs

```diff
@@ -471,6 +472,18 @@ impl fmt::Display for NonStandardSighashType {
     }
 }
 
+impl FromHexStr for NonStandardSighashType {
+    type Error = crate::parse::ParseIntError;
+
+    fn from_hex_str_no_prefix<S: AsRef<str> + Into<String>>(s: S) -> Result<Self, Self::Error>
+    where
+        Self: Sized,
+    {
+        let sighash = crate::parse::hex_u32(s)?;
+        Ok(Self(sighash))
```

Comment:
Same comment here

Reply:
NonStandardSighashType doesn't have an implementation of from_consensus